Leave/leave_interface.py

```diff
@@ -154,6 +154,23 @@ def GetReasonOfLeaves(leaves_array):
 
     return leaves_array[0]["reason"]
 
+def GetMissingMembersByRole(members_all, role, voice_channel_members):
+
+    date = datetime.datetime.strptime(datetime.datetime.today().strftime('%d/%m/%Y'), '%d/%m/%Y')
+
+    role_member_list = list(filter(lambda member: role in member.roles, members_all))
+    role_member_in_voicechannel_list = list(filter(lambda member : role in member.roles, voice_channel_members))
+    not_here_all = list(set(role_member_list) - set(role_member_in_voicechannel_list))
+
+    approved_leaves = list(filter(lambda member : (IsMemberOnLeave(member.id, date))[0], not_here_all))
+    approved_leaves_names = [member.display_name for member in approved_leaves]
+    approved_leaves_reasons = [IsMemberOnLeave(member.id, date)[1] for member in approved_leaves]
```

Comment:
Very messy. Avoid multiple arrays whose indexes share a relationship. When you find that pattern, instead go for a more extensive data structure, like a dictionary here instead.

Reply:
Given that I will be iterating over them, a dictionary would not be efficient. Instead I'll do any permutations I want within the function before I send them to the embed as a string